url3426a97ee4_rtreffer_minidns_tgz-pJ8-de_measite_minidns_minidnsrepl_MiniDnsReplJ8/src/de/measite/minidns/dnsserverlookup/AndroidUsingReflection.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015-2016 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dnsserverlookup;
+
+import de.measite.minidns.util.PlatformDetection;
+
+import java.lang.reflect.Method;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.logging.Level;
+
+/**
+ * Try to retrieve the list of DNS server by calling SystemProperties.
+ */
+public class AndroidUsingReflection extends AbstractDNSServerLookupMechanism {
+
+    public static final DNSServerLookupMechanism INSTANCE = new AndroidUsingReflection();
+    public static final int PRIORITY = 1000;
+
+    protected AndroidUsingReflection() {
+        super(AndroidUsingReflection.class.getSimpleName(), PRIORITY);
+    }
+
+    @Override
+    public String[] getDnsServerAddresses() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.